CHANGELOG.md

```diff
@@ -1,6 +1,46 @@
 # go-ipfs changelog
 
-## 0.4.12-rc2 2017-10-26
+## 0.4.12 2017-11-09
+
+Ipfs 0.4.12 brings with it many important fixes for the huge spike in network
+size we've seen this past month. These changes include the Connection Manager,
+faster batching in `ipfs add`, libp2p fixes that reduce cpu usage, and a bunch
+of new documentation.
+
+The most critical change is the 'Connection Manager', it allows an ipfs node to
+maintain a limited set of connections to other peers in the network. By default
+(and with no config changes required by the user) ipfs nodes will now try to
+maintain between 600 and 900 open connections. This is still likely higher than
+needed, and future releases may lower the default recommendation, but for now
+we want to make changes gradually. The rationale for this selection of numbers
+is as follows:
+
+- The dht routing table for a large network may rise to around 400 peers
+- Bitswap connections tend to be separate from the dht
+- PubSub connections also generally are another distinct set of peers
+  (including js-ipfs nodes)
+
+Because of this, we selected 600 as a 'LowWater' number, and 900 as a
+'HighWater' number to avoid having to clear out connections too frequently. 
+You can configure different numbers as you see fit via the `Swarm.ConnMgr`
+field in your ipfs config file. See
+[here](https://github.com/ipfs/go-ipfs/blob/master/docs/config.md#connmgr) for
+more details.
+
+Disk utilization during `ipfs add` has been optimized by doing batch writes in
```

Comment:
Technically, this currently only happens on a per-file basis. We still flush after adding each file. We should also be *adding* files in parallel (or reusing the batch but that makes it hard to report when a file has actually been added).

Probably don't need to explain this though, I just want to make sure we're all on the same page.

Reply:
If you want to take a stab at rewording this then please do.

Reply:
I've updated it. Does the new version work for you?